src/site/content/en/blog/virtualkeyboard/index.md

```diff
@@ -0,0 +1,186 @@
+may require more control over the browser's behavior. Examples include multi-screen mobile devices
+where the traditional approach would result in "wasted" screen real estate if the virtual keyboard
+is displayed on just one screen segment, but where the available viewport is shrunk on both screens
+nonetheless. The image below shows how the VirtualKeyboard API could be used to optimize the layout
+of the document dynamically to compensate for the virtual keyboard's presence.
+
+{% Img src="image/8WbTDNrhLsU0El80frMBGE4eMCD3/RHEvtaZIcCCgoWuS5c3Z.png", alt="The traditional approach results in \"wasted\" screen real estate if the virtual keyboard is displayed on just one screen segment of a multi-screen device, but where the available viewport is shrunk on both screens nonetheless.", width="800", height="301" %}
+
+Situations like this is where the VirtualKeyboard API comes in. It consists of three parts:
+
+- The `VirtualKeyboard` interface on the `navigator` object for programmatic access to the virtual
+  keyboard from JavaScript.
+- A set of CSS environment variables that provide information about the virtual keyboard's
+  appearance.
+- A virtual keyboard policy that determines if the virtual keyboard should be shown.
+
+## Feature detection and browser support
+
+To detect if the VirtualKeyboard API is supported in the current browser, use the following snippet:
+
+```js
+if ('virtualKeyboard' in navigator) {
+  // The VirtualKeyboard API is supported!
+}
+```
+
+The VirtualKeyboard API is implemented as of Chromium&nbsp;94 on desktop and mobile.
+
+## Using the VirtualKeyboard API
+
+The VirtualKeyboard API adds a new interface `VirtualKeyboard` to the `navigator` object.
+
+### Opting in to the new virtual keyboard behavior
+
+To tell the browser that you are taking care of virtual keyboard occlusions yourself, you need to
+first opt in to the new virtual keyboard behavior by setting the boolean property `overlaysContent`
+to `true`.
+
+```js
+navigator.virtualKeyboard.overlaysContent = true;
+```
+
+### Showing and hiding the virtual keyboard
+
+You can programmatically show the virtual keyboard by calling its `show()` method. For this to work,
+the focused element needs to be a form control (such as a `textarea` element), or be an editing host
+(for example, by using the
+[`contenteditable`](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/contentEditable)
+attribute). The method returns `undefined` but triggers a `geometrychanged` event.
+
+```js
+navigator.virtualKeyboard.show();
+```
+
+To hide the virtual keyboard, call the `hide()` method. The method returns `undefined` but triggers
+a `geometrychanged` event.
+
+```js
+navigator.virtualKeyboard.hide();
+```
+
+### Being informed of geometry changes
+
+Whenever the virtual keyboard appears or disappears, the `geometrychanged` event is dispatched. The
+event's `target` property contains the new geometry of the virtual keyboard inset as a
```

Comment:
I'm trying to decide if I think 'inset' needs to be defined explicitly for the reader. Can you please define it for me?

Reply:
Turns out MDN [has our back](https://developer.mozilla.org/en-US/docs/Web/CSS/inset) here.

Reply:
Worth linking to?